daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Profile.scala

```diff
@@ -37,21 +34,8 @@ final class Profile {
 }
 
 object Profile {
-  // NOTE(MH): See the documenation of [[Profile]] above for why we use
-  // [[AnyRef]] for the labels.
-  private[lf] final case class Event(val open: Boolean, val rawLabel: AnyRef, val time: Long) {
-    def label: String = {
-      import com.daml.lf.speedy.SExpr._
-      rawLabel match {
-        case null => "<null>" // NOTE(MH): We should never see this but it's no problem if we do.
-        case AnonymousClosure => "<lambda>"
-        case LfDefRef(ref) => ref.qualifiedName.toString()
-        case ChoiceDefRef(tmplRef, name) => s"exercise @${tmplRef.qualifiedName} ${name}"
-        case ref: SEBuiltinRecursiveDefinition.Reference => ref.toString().toLowerCase()
-        case str: String => str
-        case any => s"<unknown ${any}>"
-      }
-    }
+  private[lf] final case class Event(val open: Boolean, val rawLabel: Label, val time: Long) {
+    def label: String = LabelModule.Allowed renderLabel rawLabel
```

Comment:
I find the infix syntax really confusing...
```suggestion
    def label: String = LabelModule.Allowed.renderLabel(rawLabel)
```

Reply:
I feel exactly the opposite. Maybe that is down to my Smalltalk experience. But 🤷‍♂️